### PR TITLE
[Augur] Add augur tree iqtree model type to output

### LINKED ISSUE #665: [Augur] output results of ModelFinder when substitution_model = "auto"
:cool:

### :pushpin: Explain the Request

Users currently have the option within Augur_PHB to invoke ModelFinder, which will select the best substitution model and use that model to construct a phylogenetic tree. However, Augur_PHB does not output the results of ModelFinder, so users do not know which model was used to build the tree. In addition to providing crucial information to the user, this would also provide consistency across PHB workflows, as we output these files in Snippy_Streamline_PHB.

### PR DESCRIPTION
**Pull request by Michal-Babins**
This PR closes #665 

🗑️ This dev branch should <NOT> be deleted after merging to main.

## :brain: Summary

Adds a condition to augur tree to extract the model type if method is iqtree and model type is "auto', else if model is provided for iqtree extract that user provided model, else return null. A new output `iqtree_model_used` has been added to augur. 

## :zap: Impacted Workflows/Tasks

Tasks:
- `task_augur_tree.wdl`
Workflows:
- `wf_augur.wdl`
This PR may lead to different results in pre-existing outputs: **No**

This PR uses an element that could cause duplicate runs to have different results: **No**

## :hammer_and_wrench: Changes

A conditional statement has been added for task augur tree if the method is `iqtree` to either scrape the log file to extract the best fit model if `substitution_model` is `auto`, else for user provided or default GTR model assign that to the output, if iqtree is not used echo an empty string into the file. This output is captured by `iqtree_model_used` which is consistent with how snippy variant names the output. 

Note: If method is `iqtree` and substitution model is `auto`, I am having to do an awkward work around to get where the log file containing the best fit model lives. Me and @kapsakcj took a look at how augur tree treats the execution directory and it seemed to be localizing data in and out from the alignment task. More detailed review of this can be done upon request. 

### ⬅️ Outputs

Outputs added for the follwowing:
### task_augur_tree.wdl:
- `String iqtree_model_used = read_string("FINAL_MODEL.txt")`
### wf_augur.wdl:
- `String iqtree_model_used = augur_tree.iqtree_model_used`

## :test_tube: Testing

The PR was tested against  the [vrdl_vpd_specimen_HAV_set](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/data) using [this Augur_PHB](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/workflows/cdph-terrabio-taborda-manual/Augur_PHB_CDPH_VRDL_settings_HAV) instance. The `augur_tree.substitution_model` was tested with model set to [auto](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/72c99d41-bad2-472c-84a4-55f01b70fbec), with model manually set to [TIM2](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/252156fe-2253-4d80-bb8b-b20ac03dde90), with model manually set to [GTR](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/74fe7e8d-be0d-46a9-bec6-5bf48c1ebfc8), with model manually set to [F81](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/job_history/e532a37c-0949-4b5f-b176-e74a1827333f). The null object pattern was tested with `augur_tree.substitution_model` unset and `augur_tree.method` set to `fasttree`.

### Suggested Scenarios for Reviewer to Test

Please test with this workflow configured [here](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Bioinformatics_Development/workflows/cdph-terrabio-taborda-manual/Augur_PHB_CDPH_VRDL_settings_HAV) using the `mb-augur-automodel-output-dev` branch. Test with `augur_tree.substitution_model`  set to `auto` and a few models manually set to your choosing from this [set](http://www.iqtree.org/doc/Substitution-Models). Check to see iqtree_model_used on the set has been updated appropriately.  Also please test without using `iqtree` as the `augur_tree.method` (fasttree or raxml).  Check to see `iqtree_model_used` is null.

## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [x] All changed results have been confirmed
- [x] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [x] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments
- [x] The documentation has been updated

### CODE REVIEW

File: docs/workflows/phylogenetic_construction/augur.md [hunk 1]
```diff
@@ -290,6 +290,7 @@ The Nextstrain team hosts documentation surrounding the Augur workflow → Auspi
 | auspice_input_json | File | JSON file used as input to Auspice |
 | combined_assemblies | File | Concatenated FASTA file containing all samples |
 | distance_tree | File | The distance tree created in Newick (.nwk) format |
+| iqtree_model_used | String | The iqtree model used during augur tree |
```
Comment (sage-wright) [inline docs/workflows/phylogenetic_construction/augur.md:293]: can you provide that link to the iqtree documentation page that lists the models available?

probalby in the inputs, not the outptus

File: workflows/phylogenetics/wf_augur.wdl [hunk 2]
```diff
@@ -203,6 +203,7 @@ workflow augur {
     File? auspice_input_json = augur_export.auspice_json
     File? time_tree = augur_refine.refined_tree
     File distance_tree = augur_tree.aligned_tree
+    String iqtree_model_used = augur_tree.iqtree_model_used
```
[the remark this answers is not in the document]
Comment (Michal-Babins) [reply workflows/phylogenetics/wf_augur.wdl:206]: It was pointed out to me to reference it how snippy used it here:
https://github.com/theiagen/public_health_bioinformatics/blob/main/tasks/phylogenetic_inference/task_iqtree2.wdl#L82C5-L82C65

File: docs/workflows/phylogenetic_construction/augur.md (hunk 1 above)
Comment (Michal-Babins) [reply docs/workflows/phylogenetic_construction/augur.md:293]: http://www.iqtree.org/doc/Substitution-Models

Comment (sage-wright) [reply docs/workflows/phylogenetic_construction/augur.md:293]: awesome! can you add that to the docs, in the inputs?

File: workflows/phylogenetics/wf_augur.wdl (hunk 2 above)
[the remark this answers is not in the document]
Comment (sage-wright) [reply workflows/phylogenetics/wf_augur.wdl:206]: only thought about changing this is that if we run both those workflows on the same set they'll overwrite

[the remark this answers is not in the document]
Comment (Michal-Babins) [reply workflows/phylogenetics/wf_augur.wdl:206]: Got it, I'll change it to augur_iqtree_model_used at the workflow level

File: docs/workflows/phylogenetic_construction/augur.md (hunk 1 above)
Comment (Michal-Babins) [reply docs/workflows/phylogenetic_construction/augur.md:293]: Done. Let me know if you like the formatting.

Comment (sage-wright) [reply docs/workflows/phylogenetic_construction/augur.md:293]: perf, thanks

Comment (sage-wright) [inline docs/workflows/phylogenetic_construction/augur.md:293]: just update this doc to be augur_iqtree_model_used and then i'll merge 👍

Comment (Michal-Babins) [reply docs/workflows/phylogenetic_construction/augur.md]: Good catch

**Review by sage-wright — APPROVED** [review]
well done. ⭐

**Pull request merged**